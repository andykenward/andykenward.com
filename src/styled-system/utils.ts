import shouldForwardProp from "@styled-system/should-forward-prop"
import { equals, indexOf } from "ramda"

import { colorPropsKeys } from "./colors"
import { fontSizePropsKeys } from "./fonts"

const propKeys = [...colorPropsKeys, ...fontSizePropsKeys]

export const filterProps: (prop: string) => boolean = (prop: string): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propKeys), -1)
