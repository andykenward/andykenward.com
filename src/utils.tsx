import shouldForwardProp from "@styled-system/should-forward-prop"
import { ThemeProvider } from "emotion-theming"
import { equals, indexOf } from "ramda"
import React from "react"
import { render } from "@testing-library/react"

import { theme } from "./theme"

export const filterProps: (prop: string, propsToFilter: string[]) => boolean = (
  prop: string,
  propsToFilter: string[]
): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propsToFilter), -1)

export const childrenOnly: (prop: string) => boolean = (
  prop: string
): boolean => prop === "children" || prop === "data-test-id"

export const renderWithTheme = (element: React.ReactElement<any>) =>
  render(<ThemeProvider theme={theme}>{element}</ThemeProvider>)
