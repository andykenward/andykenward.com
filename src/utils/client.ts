import { GraphQLClient } from "graphql-request"

import { getSdk } from "../../generated/graphql"

const graphqlClient = new GraphQLClient(process.env.API_URL, {
  headers: {
    authorization: `Bearer ${process.env.API_TOKEN}`,
  },
})

export const appSdk = (() => getSdk(graphqlClient))()
