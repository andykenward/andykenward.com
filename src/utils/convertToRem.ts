import { is } from "ramda"
/**
 *
 * UTILS
 */

export const convertToRem = (n: number | string): string | number =>
  is(Number, n) && n != 0 ? `${(n as number) / 16}rem` : n
