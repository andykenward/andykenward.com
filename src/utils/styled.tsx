import styled, { CreateStyled } from "@emotion/styled"
import React from "react"

import {
  CSSObject,
  SerializedStyles,
  InterpolationWithTheme,
} from "@emotion/core"

import { Theme } from "../theme"

/**
 * TODO: pass `as` typings down if component
 */
export interface StyledProps {
  as?: keyof JSX.IntrinsicElements | React.ComponentType<any>
  css?: CSSObject | SerializedStyles | InterpolationWithTheme<any>
}

export default styled as CreateStyled<Theme>
