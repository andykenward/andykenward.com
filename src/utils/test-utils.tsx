import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { ThemeProvider } from "theme-ui"

import { theme } from "../theme"

const AllTheProviders: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllTheProviders, ...options })

export * from "@testing-library/react"

export { customRender as render }
