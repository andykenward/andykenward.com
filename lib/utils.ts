import shouldForwardProp from "@styled-system/should-forward-prop"
import { equals, indexOf } from "ramda"

import { propKeys } from "./styled-system"

export const filterProps: (prop: string) => boolean = (prop: string): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propKeys), -1)

export const childrenOnly: (prop: string) => boolean = (
  prop: string
): boolean => prop === "children" ?? prop === "data-test-id"
