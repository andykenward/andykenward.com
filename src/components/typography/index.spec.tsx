import { matchers } from "jest-emotion"
import React from "react"

import { renderWithTheme } from "../../utils"

import {
  HeadingOne,
  HeadingTwo,
  HeadingThree,
  HeadingFour,
  BodyBold,
  Body,
  Label,
} from "."

// Add the custom matchers provided by 'jest-emotion'
expect.extend(matchers)

describe("Typography", () => {
  describe("<HeadingOne />", () => {
    it("renders correctly", () => {
      const { asFragment, container } = renderWithTheme(
        <HeadingOne>Heading One</HeadingOne>
      )
      expect(container.firstChild).toHaveStyleRule(
        "font-family",
        `"Cooper Light",Georgia,Times,"Times New Roman",serif`
      )
      expect(container.firstChild).toHaveStyleRule("font-size", "44px")
      expect(container.firstChild).toHaveStyleRule("font-weight", "300")
      expect(container.firstChild).toHaveStyleRule("line-height", "1.090909091")
      expect(container.firstChild).toHaveStyleRule("color", "rgb(36,28,21)")
      expect(container.firstChild).toHaveStyleRule("margin", "0")
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe("<HeadingTwo />", () => {
    it("renders correctly", () => {
      const { asFragment, container } = renderWithTheme(
        <HeadingTwo>Heading Two</HeadingTwo>
      )
      expect(container.firstChild).toHaveStyleRule(
        "font-family",
        `Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif`
      )
      expect(container.firstChild).toHaveStyleRule("font-size", "36px")
      expect(container.firstChild).toHaveStyleRule("font-weight", "400")
      expect(container.firstChild).toHaveStyleRule("letter-spacing", "-0.5px")
      expect(container.firstChild).toHaveStyleRule("line-height", "1.25")
      expect(container.firstChild).toHaveStyleRule("color", "rgb(36,28,21)")
      expect(container.firstChild).toHaveStyleRule("margin", "0")
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe("<HeadingThree />", () => {
    it("renders correctly", () => {
      const { asFragment, container } = renderWithTheme(
        <HeadingThree>Heading Three</HeadingThree>
      )
      expect(container.firstChild).toHaveStyleRule(
        "font-family",
        `Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif`
      )
      expect(container.firstChild).toHaveStyleRule("font-size", "28px")
      expect(container.firstChild).toHaveStyleRule("font-weight", "400")
      expect(container.firstChild).toHaveStyleRule("letter-spacing", "-0.2px")
      expect(container.firstChild).toHaveStyleRule("line-height", "1.25")
      expect(container.firstChild).toHaveStyleRule("color", "rgb(36,28,21)")
      expect(container.firstChild).toHaveStyleRule("margin", "0")
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe("<HeadingFour />", () => {
    it("renders correctly", () => {
      const { asFragment, container } = renderWithTheme(
        <HeadingFour>Heading Four</HeadingFour>
      )
      expect(container.firstChild).toHaveStyleRule(
        "font-family",
        `Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif`
      )
      expect(container.firstChild).toHaveStyleRule("font-size", "20px")
      expect(container.firstChild).toHaveStyleRule("font-weight", "400")
      expect(container.firstChild).toHaveStyleRule("line-height", "1.25")
      expect(container.firstChild).toHaveStyleRule("color", "rgb(36,28,21)")
      expect(container.firstChild).toHaveStyleRule("margin", "0")
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe("<BodyBold />", () => {
    it("renders correctly", () => {
      const { asFragment, container } = renderWithTheme(
        <BodyBold>Body Bold</BodyBold>
      )
      expect(container.firstChild).toHaveStyleRule(
        "font-family",
        `Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif`
      )
      expect(container.firstChild).toHaveStyleRule("font-size", "16px")
      expect(container.firstChild).toHaveStyleRule("font-weight", "500")
      expect(container.firstChild).toHaveStyleRule("line-height", "1.5")
      expect(container.firstChild).toHaveStyleRule("color", "rgb(36,28,21)")
      expect(container.firstChild).toHaveStyleRule("margin", "0")
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe("<Body />", () => {
    it("renders correctly", () => {
      const { asFragment, container } = renderWithTheme(<Body>Body</Body>)
      expect(container.firstChild).toHaveStyleRule(
        "font-family",
        `Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif`
      )
      expect(container.firstChild).toHaveStyleRule("font-size", "16px")
      expect(container.firstChild).toHaveStyleRule("font-weight", "400")
      expect(container.firstChild).toHaveStyleRule("line-height", "1.5")
      expect(container.firstChild).toHaveStyleRule("color", "rgb(36,28,21)")
      expect(container.firstChild).toHaveStyleRule("margin", "0")
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe("<Label />", () => {
    it("renders correctly", () => {
      const { asFragment, container } = renderWithTheme(<Label>Label</Label>)
      expect(container.firstChild).toHaveStyleRule(
        "font-family",
        `Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif`
      )
      expect(container.firstChild).toHaveStyleRule("font-size", "14px")
      expect(container.firstChild).toHaveStyleRule("font-weight", "500")
      expect(container.firstChild).toHaveStyleRule("line-height", "1.5")
      expect(container.firstChild).toHaveStyleRule("color", "rgb(36,28,21)")
      expect(container.firstChild).toHaveStyleRule("margin", "0")
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
