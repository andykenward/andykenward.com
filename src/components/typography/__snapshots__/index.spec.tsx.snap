// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography <Body /> renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "Open Sans",serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  color: rgb(36,28,21);
  margin: 0;
}

<p
    class="emotion-0"
  >
    Body
  </p>
</DocumentFragment>
`;

exports[`Typography <BodyBold /> renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "Open Sans",serif;
  font-size: 16px;
  font-weight: 600;
  line-height: 1.5;
  color: rgb(36,28,21);
  margin: 0;
}

<p
    class="emotion-0"
  >
    Body Bold
  </p>
</DocumentFragment>
`;

exports[`Typography <HeadingFour /> renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "Open Sans",serif;
  font-size: 20px;
  font-weight: 400;
  line-height: 1.25;
  color: rgb(36,28,21);
  margin: 0;
}

<h4
    class="emotion-0"
  >
    Heading Four
  </h4>
</DocumentFragment>
`;

exports[`Typography <HeadingOne /> renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: Merriweather,sans-serif;
  font-size: 44px;
  font-weight: 300;
  line-height: 1.090909091;
  color: rgb(36,28,21);
  margin: 0;
}

<h1
    class="emotion-0"
  >
    Heading One
  </h1>
</DocumentFragment>
`;

exports[`Typography <HeadingThree /> renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "Open Sans",serif;
  font-size: 28px;
  font-weight: 400;
  -webkit-letter-spacing: -0.2px;
  -moz-letter-spacing: -0.2px;
  -ms-letter-spacing: -0.2px;
  letter-spacing: -0.2px;
  line-height: 1.25;
  color: rgb(36,28,21);
  margin: 0;
}

<h3
    class="emotion-0"
  >
    Heading Three
  </h3>
</DocumentFragment>
`;

exports[`Typography <HeadingTwo /> renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "Open Sans",serif;
  font-size: 36px;
  font-weight: 400;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  line-height: 1.25;
  color: rgb(36,28,21);
  margin: 0;
}

<h2
    class="emotion-0"
  >
    Heading Two
  </h2>
</DocumentFragment>
`;
